### ISSUE TITLE
General CLI improvements

### ISSUE THREAD

**Issue by mrkldshv** [ds300/lazyrepo#2]
I was looking through the source code and was thinking that it might be a good idea to use some lightweight framework for building CLI. It should make it easier to maintain current and add new commands in the future. It would also simplify things like help command, arguments parsing, validation etc.

Please, let me know your thoughts about it.

**Comment by ds300** [comment]
Yeah I'd love this. Was thinking about it yesterday but decided to kick the can down the road. 

The only part of the CLI that I want to stay stable at this stage is the core `lazy <script-name> [...<forward-args>]` thing. It's critical that folks don't need to worry about name clashes with lazy's internal commands, or adding a '--' separator to forward args to their scripts.

Happy for you to look into adding this if you're interested.

**Comment by mrkldshv** [comment]
Thanks for reply! I'll look into adding some simple solution. Will submit PR where we can discuss possible options, or reply here if I have any questions or concerns.

**Comment by mrkldshv** [comment]
I'm currently working on that, and thinking of some CLI changes. Here's the help message of the root command:
```
Usage:
  $ lazyrepo <task> [...forward-args]

Commands:
  <task> [...forward-args]          run task in all packages
  :run <task> -- [...forward-args]  run task in all packages
  :init                             create config file
  :clean                            delete all local cache data
  :inherit                          run command from configuration file specified by script name

Options:
  --filter <paths>  [string] run task in packages specified by paths 
  --force           [boolean] ignore existing cached artifacts 
```
Changes:
* make `:force` command an option
* make `[...filter-paths]` arguments an option

Let me know what you think about that!

**Comment by ds300** [comment]
Yeah this feels really good! No notes :)

**Comment by mrkldshv** [comment]
Great! Will submit PR with these changes shortly :)

**Comment by ds300** [comment]
I've been thinking about this some more, after using lazyrepo a bunch the last few days, and I'm starting to think the colon-prefix thing is a bad way to go. It's unusual and even I'm finding it difficult to get used to.

Also I'm no longer sure the requirement of being able to forward args without `--` makes sense.  
I suspect we could make it required and supply a good error message in case the user passes an unrecognized arg.

```bash
$ lazy test --runInBand

❗ Option '--runInBand' not recognized.

💡 Did you mean to run `lazy test -- --runInBand`?
```

After which I think it's fine to define our commands without colons and then add helpful tips for when conflicts are detected.

e.g. say the user has a package.json like

```json
"scripts": {
  "clean": "rm -rf dist"
}
```

and then they run `lazy clean`, it could be like:

```bash
$ lazy clean

Clearing 13 .lazy cache files...  Done!

💡 If you meant to call your "clean" npm script, run

   lazy run clean

```

What do you think? This is more in line with how tools like `pnpm` and `turbo` work, so it feels like a safer bet to me.

**Comment by mrkldshv** [comment]
All this sounds good to me! 

Actually, during the CLI refactoring I had to do a few hacks to forward arguments without `--` for the root command. So, I'm glad we decided to make this change. It will simplify a few things in the code and keep the same experience across tools and our own CLI (root and run commands).

I'm also with you on removing colon prefix for the commands. I was constantly forgetting to prefix commands when using lazyrepo. I really like your suggestions to provide helpful error messages. I'll check if any of CLI frameworks can do any of these out of the box, and, if not, we can handle it ourselves.

I'm going to submit PR with POC to get some feedback and suggestions. It's not easy to find CLI framework that meets all our requirements, so I'd like to discuss possible tradeoffs in this area.
